OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -39,6 +39,17 @@ public abstract class AttackBaseInfo : PausableConditionalTraitInfo
 
 		[VoiceReference] public readonly string Voice = "Action";
 
+		[Desc("Tolerance for attack angle. Range [0, 128], 128 covers 360 degrees.")]
+		public readonly int FacingTolerance = 128;
```

Comment:
It looks like we're going to need at least an update rule that sets `FacingTolerance: 0` on all `AttackFrontal` and `FacingTolerance: 2` on `AttackBomber` if not already set to another value (and then prompt the user to check these definitions because some may have been made in error).

But this raises a bigger question: if `AttackFrontal` defaults to being able to attack from any angle, then why is it still called `AttackFrontal` and how does it differ from `AttackOmni`?

Reply:
I was a bit doubtful myself whether `FacingTolerance` should default to 0 or 128. But if it defaults to 0 we would need an update rule for `AttackFollow`, so same difference, I guess?

With regards to the difference between `AttackFrontal` and `AttackOmni:` All the various Attack* activities are a bit of a mess in my opinion and I intend to do a follow up to consolidate and sanitize them a bit. As it stands, the most poignant difference has nothing to do with their facing, but rather with movement. `AttackFollow` can move while attacking, `AttackFrontal` must stop before attacking and `AttackOmni` has no provision for movement at all. So I guess they should be named something like 'AttackStop' and 'AttackStatic'. I don't want to push the scope of this PR even further than it already has though, but rather address these and related general attack refactors in a new PR.

Keeping that in mind, I'm not really keen on foisting a complicated update rule on modders if these parameters might change again soon anyway. Maybe a stop-gap solution would be to have `AttackFrontal` and `AttackBomber` override the parameter defaults, so as to minimize the impact on modders while further refactors are pending? I don't know how to do that without breaking the code though.

Reply:
Would it be feasible to pass the facing tolerance in as an argument to `TargetInFiringArc`, and then keep it defined on the subclasses instead of the parent `AttackBase` for now? This may be the simplest way to defer the problem to future, more appropriate PRs.

Reply:
Implemented the `FacingTolerance` overrides on `AttackFrontal` and `AttackBomber` and it seems to work now.